tests/unit/amo/components/TestAddonRecommendations.js

```diff
@@ -0,0 +1,224 @@
+        const randomizer = fakeRandomizer(expectedRecommendedValue);
+        render({ cookie, randomizer });
+        sinon.assert.calledWith(cookie.save,
+          TAAR_COHORT_COOKIE_NAME, expectedCohort, { path: '/' });
+      };
+
+    checkWithRandomizerValue(true, TAAR_COHORT_INCLUDED);
+    checkWithRandomizerValue(false, TAAR_COHORT_EXCLUDED);
+  });
+
+  it('should dispatch a fetch action if no recommendations exist', () => {
+    const addon = createInternalAddon(fakeAddon);
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+    const errorHandler = createStubErrorHandler();
+    const randomizer = fakeRandomizer(true);
+
+    render({ addon, errorHandler, randomizer });
+
+    sinon.assert.calledWith(dispatchSpy, fetchRecommendations({
+      errorHandlerId: errorHandler.id,
+      guid: addon.guid,
+      recommended: true,
+    }));
+  });
+
+  it('should not dispatch a fetch action if addon is null', () => {
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+
+    render({ addon: null });
+
+    sinon.assert.notCalled(dispatchSpy);
+  });
+
+  it('should dispatch a fetch action if the addon is updated', () => {
+    const addon = createInternalAddon(fakeAddon);
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+    const errorHandler = createStubErrorHandler();
+    const randomizer = fakeRandomizer(true);
+
+    const root = render({ addon, errorHandler, randomizer });
+
+    dispatchSpy.reset();
+
+    const newAddon = createInternalAddon({
+      ...fakeAddon,
+      guid: 'new-guid',
+    });
+
+    root.setProps({ addon: newAddon });
+
+    sinon.assert.calledWith(dispatchSpy, fetchRecommendations({
+      errorHandlerId: errorHandler.id,
+      guid: newAddon.guid,
+      recommended: true,
+    }));
+  });
+
+  it('should not dispatch a fetch if the addon is updated but not changed', () => {
+    const addon = createInternalAddon(fakeAddon);
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+
+    const root = render({ addon });
+
+    dispatchSpy.reset();
+
+    root.setProps({ addon });
+
+    sinon.assert.notCalled(dispatchSpy);
+  });
+
+  it('should not dispatch a fetch if the addon is updated to null', () => {
+    const addon = createInternalAddon(fakeAddon);
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+
+    const root = render({ addon });
+
+    dispatchSpy.reset();
+
+    root.setProps({ addon: null });
+
+    sinon.assert.notCalled(dispatchSpy);
+  });
+
+  it('should send a GA ping when recommendations are loaded', () => {
+    const { fallbackReason, outcome } = fakeRecommendations;
+    const root = render({ tracking: fakeTracking });
+
+    root.setProps({ recommendations: fakeRecommendations });
+
+    sinon.assert.calledOnce(fakeTracking.sendEvent);
+    sinon.assert.calledWith(fakeTracking.sendEvent, {
+      action: `outcome: ${outcome} | fallbackReason: ${fallbackReason}`,
+      category: TAAR_IMPRESSION_CATEGORY,
+      label: fakeAddon.name,
+    });
+  });
```

Comment:
I think we need a test for the early `loading` return in case we have a recommendation but it is still loading.